### PR TITLE
feat(esl-popup): couple new features and refactoring

### PR DESCRIPTION
**Pull request by dshovchko**
In scope of this PR:
 - added `disable-activator-observation` attribute to let popup leaves viewport without hiding
 - added new `fit-on-major` value for the behavior attribute so that the popup update its position in order to fit on the major axis only
 - added new `fit-on-minor` value for the behavior attribute so that the popup update its position in order to fit on the minor axis only
 - added RTL support, resulting in a popup mirrored relative to the major axis
 - updated TSX shapes
 - refactor code to achieve the best performance during popup position updating

### CODE REVIEW

File: src/modules/esl-popup/core/esl-popup.ts [hunk 1]
```diff
@@ -91,32 +97,36 @@ export class ESLPopup extends ESLToggleable {
...
   }
 
-  /** Is position along vertical axis? */
-  protected get _isPositioningAlongVertical() {
-    return ['top', 'bottom'].includes(this.position);
+  /** Checks that the position along the vertical axis */
+  @memoize()
+  protected get _isMajorAxisVertical(): boolean {
+    return !isMajorAxisHorizontal(this.position);
   }
```
Comment (ala-n) [inline src/modules/esl-popup/core/esl-popup.ts:115]: Cosmetic: it looks like the memoize will be much heavier here then exact check inside the `isMajorAxisHorizontal`

**Review by ala-n — APPROVED** [review]

File: src/modules/esl-popup/core/esl-popup.ts (hunk 1 above)
Comment (dshovchko) [reply src/modules/esl-popup/core/esl-popup.ts:115]: I was of the same opinion at first.
But I wondered to check and understand with the help of benchmarks. I don't seem to have the same understanding now. 
We can discuss it.

**Pull request merged**